mesonbuild/compilers/mixins/clike.py

```diff
@@ -688,10 +690,21 @@ def get_define(self, dname: str, prefix: str, env: 'Environment',
             cached = p.cached
             if p.returncode != 0:
                 raise mesonlib.EnvironmentException(f'Could not get define {dname!r}')
-        # Get the preprocessed value after the delimiter,
-        # minus the extra newline at the end and
-        # merge string literals.
-        return self._concatenate_string_literals(p.stdout.split(delim + '\n')[-1][:-1]).strip(), cached
+
+        # Get the preprocessed value between the delimiters
+        star_idx = p.stdout.find(delim_start)
+        end_idx = p.stdout.rfind(delim_end)
+        if (star_idx == -1) or (end_idx == -1) or (star_idx == end_idx):
+            raise AssertionError('BUG: Delimiters not found in preprocessor output!')
```

Comment:
BTW: this should really be a MesonBugException.

Reply:
Oh ok sure, I have lost track of all the kinds of different exceptions and when to use which 😅